### PR TITLE
[CVP-2062] implementing rehearsal part

### PR DESCRIPTION
**Pull request by jsztuka**

### CODE REVIEW

**Comment by jsztuka** [comment]
Created generic claim without any jobs yet. Going to include cvp-specific steps.

File: ci-operator/config/redhat-operator-ecosystem/cvp/redhat-operator-ecosystem-cvp-ocp-4.7.yaml [hunk 1]
```diff
@@ -11,6 +11,44 @@ resources:
...
+      BASE_DOMAIN: cpaas-ci.devcluster.openshift.com
+      CHANNEL: "4.3"
+      INSTALL_NAMESPACE: '!create'
+      PACKAGE: e2e-test-operator
+      PYXIS_URL: ""
+      TARGET_NAMESPACES: '!install'
+    post:
+    - chain: ipi-aws-post
+    pre:
+    - chain: ipi-aws-pre
```
Comment (petr-muller) [inline ci-operator/config/redhat-operator-ecosystem/cvp/redhat-operator-ecosystem-cvp-ocp-4.7.yaml:36]: No, this does not make sense. `ipi-aws-pre` and `ipi-aws-post` are chains that contain steps that install and deprovision clusters, they will not work with claimed clusters. What you need to do is to take what the `generic-claim` workflow has in `pre` and `post`, and simply add CVP-specific steps to it (similar to how the existing CVP workflows just add more steps to what e.g. `ipi-aws` has).

Basically just take all the CVP steps here and move them to the workflow. Then the test here will not need to specify any pre/test/post, exactly how the other tests (that use the existing workflows) do not set them.

File: ci-operator/config/redhat-operator-ecosystem/cvp/redhat-operator-ecosystem-cvp-ocp-4.7.yaml [hunk 2]
```diff
@@ -11,6 +11,44 @@ resources:
...
+    - ref: optional-operators-configure-proxy-registry
+    - ref: optional-operators-cvp-common-apply-secrets
+    - ref: optional-operators-subscribe
+    test:
+    - ref: optional-operators-cvp-common-scorecard
+    - ref: optional-operators-cvp-common-test
+    - as: claim
+      commands: "printenv KUBECONFIG\nprintenv KUBEADMIN_PASSWORD_FILE\noc get node\noc
+        config view\noc whoami  \n"
+      from: ocp-4.7.0-amd64
```
Comment (petr-muller) [inline ci-operator/config/redhat-operator-ecosystem/cvp/redhat-operator-ecosystem-cvp-ocp-4.7.yaml:46]: This si supposed to be an image, and there's no `ocp-4.7.0-amd64` image.

File: ci-operator/step-registry/optional-operators/cvp-common/generic-claim/OWNERS [hunk 3]
```diff
@@ -0,0 +1,7 @@
+approvers:
+  - dirgim
+  - petr-muller
```
Comment (petr-muller) [inline ci-operator/step-registry/optional-operators/cvp-common/generic-claim/OWNERS:3]: Remove me from here please

File: ci-operator/step-registry/optional-operators/cvp-common/generic-claim/optional-operators-cvp-common-generic-claim-workflow.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+workflow:
+  as: optional-operators-cvp-common-generic-claim
```
Comment (petr-muller) [inline ci-operator/step-registry/optional-operators/cvp-common/generic-claim/optional-operators-cvp-common-generic-claim-workflow.yaml:2]: It's no longer a "generic" workflow, could be called just `optional-operators-cvp-common-claim` IMO

File: ci-operator/config/redhat-operator-ecosystem/cvp/redhat-operator-ecosystem-cvp-ocp-4.7.yaml (hunk 1 above)
Comment (jsztuka) [reply ci-operator/config/redhat-operator-ecosystem/cvp/redhat-operator-ecosystem-cvp-ocp-4.7.yaml:36]: Thats exactly what I did previously and it started to fail on the same exact thing as I was mentioning earlier yesterday.

File: ci-operator/step-registry/optional-operators/cvp-common/generic-claim/optional-operators-cvp-common-generic-claim-workflow.yaml (hunk 4 above)
Comment (jsztuka) [reply ci-operator/step-registry/optional-operators/cvp-common/generic-claim/optional-operators-cvp-common-generic-claim-workflow.yaml:2]: Renamed.

File: ci-operator/step-registry/optional-operators/cvp-common/generic-claim/OWNERS (hunk 3 above)
Comment (jsztuka) [reply ci-operator/step-registry/optional-operators/cvp-common/generic-claim/OWNERS:3]: Removed.

File: ci-operator/config/redhat-operator-ecosystem/cvp/redhat-operator-ecosystem-cvp-ocp-4.7.yaml (hunk 2 above)
Comment (jsztuka) [reply ci-operator/config/redhat-operator-ecosystem/cvp/redhat-operator-ecosystem-cvp-ocp-4.7.yaml:46]: I'm not sure if I get this part correctly, where do I get the image?

File: ci-operator/config/redhat-operator-ecosystem/cvp/redhat-operator-ecosystem-cvp-ocp-4.7.yaml (hunk 1 above)
Comment (jsztuka) [reply ci-operator/config/redhat-operator-ecosystem/cvp/redhat-operator-ecosystem-cvp-ocp-4.7.yaml]: Yes, that was exactly the approach I did yesterday and it was failing on unbound variables.

Comment (jsztuka) [reply ci-operator/config/redhat-operator-ecosystem/cvp/redhat-operator-ecosystem-cvp-ocp-4.7.yaml]: But lets see how this one goes.

Comment (jsztuka) [reply ci-operator/config/redhat-operator-ecosystem/cvp/redhat-operator-ecosystem-cvp-ocp-4.7.yaml]: I see, as you said 
ipi-aws-pre and ipi-aws-post are chains that contain steps that install and deprovision clusters, they will not work with claimed clusters!!!!
This was probably causing the issue to occur. Lets try it without those chains.

File: ci-operator/step-registry/optional-operators/cvp-common/claim/optional-operators-cvp-common-claim-workflow.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+workflow:
+  as: optional-operators-cvp-common-claim
+  steps:
+    pre:
+    - ref: optional-operators-configure-proxy-registry
+    - ref: optional-operators-cvp-common-apply-secrets
+    - ref: optional-operators-subscribe
+    test:
+    - ref: optional-operators-cvp-common-scorecard
+    - ref: optional-operators-cvp-common-test
```
Comment (petr-muller) [inline ci-operator/step-registry/optional-operators/cvp-common/claim/optional-operators-cvp-common-claim-workflow.yaml:10]: So previously you had more steps here and then you have removed them. They were necessary, put them back. The resulting workflow should be basically `generic-claim` + everything CVP needs

Comment (jsztuka) [reply ci-operator/step-registry/optional-operators/cvp-common/claim/optional-operators-cvp-common-claim-workflow.yaml:10]: added those ones from generic claim

**Pull request merged**